proposals/0303-swiftpm-extensible-build-tools.md

```diff
@@ -864,11 +1038,37 @@ It is an area we are interested in exploring and improving in the near future, s
+For example, a situation in which a build tool needs [Yams](https://github.com/jpsim/Yams) v3.x while a package client needs v4.x would not be supported by this proposal, even though it would pose no real problem at runtime. It would be very desirable to support this, but that would require significant work on SwiftPM to allow it to keep multiple independent package graphs. It should be noted that if a package vends both a build tool and a runtime library that clients using the build tool must link against, then even this apparently simple case would get more complicated. In this case the runtime library would have to use Yams v4.x in order to be usable by the client package, even if the tool itself used Yams v3.x.
+
+It should also be noted that this is no different from a package that defines an executable and a library and would like to use a particular version of Yams without requiring the client of the library to use a compatible version. This is not supported in SwiftPM today either, and it is in fact another manifestation of the same problem.
+
+Solving this mixture of build-time and run-time package dependencies is possible but not without significant work in SwiftPM (and IDEs that use libSwiftPM). We think that this proposal could be extended to support declaring such dependencies in the future, and that even with this restriction, this proposal provides useful functionality for packages.
+
+### A More Sophisticated `Path` Type
+
+The current `Path` type provided in the `PackagePlugin` API is intentionally kept minimal, and should be sufficient for the needs of most plugins. A future version of this proposal would extend this type, possibly aligning it with `SwiftSystem`'s `FilePath` type. The initial API has purposefully been kept the same as `FilePath` to make such a transition easier.
+
+An alternate direction would be to replace it with a more domain-specific representation that could also keep track of the path root in relation to the directories that matter to packages and build systems, avoiding the need to form absolute paths.
+
+Since the API that is available to the plugin script is based on the tools version of the package that contains it, existing plugin scripts are expected to be able to run without modification even if the API does change.
+
+### Contextual Information About the Target Platform
+
+The current `TargetBuildContext` type in the `PackagePlugin` API provides minimal information about the structure and configuration of the target, but it does not in this initial proposal provide any information about the target platform for which the package is being built.
+
+This would be needed in order to implement more advanced tools such as code generators or linkers.
+
+### Specific Support for Code Formatters
```

Comment:
IMO this applies more to linters than formatters, as the latter modify code while the former emit information

Reply:
I'd thought of them as the same but that distinction makes sense.  I'll update it.

Reply:
Actually, separately I wrote:

> /// Another possible capability could be specific support
> /// for code linters that could emit structured diagnostics with fix-its,
> /// or for code formatters that can modify the source code as a separate
> /// action outside the build.

So I guess some part of my brain did make that distinction.